### PR TITLE
Add variables to the new API

### PR DESCRIPTION
**Pull request by fpagnoux**
Connected to #125 
Depends on https://github.com/openfisca/openfisca-dummy-country/pull/6
Depends on https://github.com/openfisca/openfisca-core/pull/494

### CODE REVIEW

**Comment by fpagnoux** [comment]
A few exemples : 

## Input variable
```JSON
{
    "defaultValue": "1970-01-01",
    "definitionPeriod": "eternity",
    "description": "Date de naissance",
    "entity": "individu",
    "references": ["https://fr.wikipedia.org/wiki/Date_de_naissance"],
    "source": "https://github.com/openfisca/openfisca-dummy-country/blob/0.1.5/openfisca_dummy_country/model/model.py#L27-L32",
    "valueType": "Date"
}/variable/salaire_net
```

## Basic variable with a formula

```JSON
{
    "defaultValue": 0,
    "definitionPeriod": "month",
    "description": "Salaire net",
    "entity": "individu",
    "formulas": {
        "0001-01-01": {
            "content": "def function(individu, period):\n    salaire_brut = individu('salaire_brut', period)\n\n    return salaire_brut * 0.8\n",
            "source": "https://github.com/openfisca/openfisca-dummy-country/blob/0.1.5/openfisca_dummy_country/model/model.py#L172-L175"
        }
    },
    "source": "https://github.com/openfisca/openfisca-dummy-country/blob/0.1.5/openfisca_dummy_country/model/model.py#L164-L175",
    "valueType": "Float"
}
```

**Comment by fpagnoux** [comment]
## Interrupted dated variable

```JSON
{
    "defaultValue": 0,
    "definitionPeriod": "month",
    "description": "Allocationd pour Parent Isolé",
    "entity": "famille",
    "formulas": {
        "2000-01-01": {
            "content": "def function_2000(famille, period):\n    nb_parents = famille.nb_persons(role = famille.PARENT)\n    nb_enfants = famille.nb_persons(role = famille.ENFANT)\n    condition = (nb_parents == 1) * (nb_enfants > 0)\n\n    return condition * 100\n",
            "source": "https://github.com/openfisca/openfisca-dummy-country/blob/0.1.5/openfisca_dummy_country/model/model.py#L229-L234"
        },
        "2005-01-01": {
            "content": "def function_2005(famille, period):\n    nb_parents = famille.nb_persons(role = famille.PARENT)\n    nb_enfants = famille.nb_persons(role = famille.ENFANT)\n    condition = (nb_parents == 1) * (nb_enfants > 0)\n\n    return condition * 200\n",
            "source": "https://github.com/openfisca/openfisca-dummy-country/blob/0.1.5/openfisca_dummy_country/model/model.py#L221-L226"
        },
        "2010-01-01": null
    },
    "source": "https://github.com/openfisca/openfisca-dummy-country/blob/0.1.5/openfisca_dummy_country/model/model.py#L212-L234",
    "valueType": "Float"
}
```

**Comment by fpagnoux** [comment]
Les liens github marcheront quand https://github.com/openfisca/openfisca-dummy-country/pull/6 sera mergée et publiée.

**Comment by fpagnoux** [comment]
I'm adding an `id` property, as for the parameters.

File: openfisca_web_api/app.py [hunk 1]
```diff
@@ -18,31 +18,40 @@ def create_app(country_package = os.environ.get('COUNTRY_PACKAGE')):
...
+    @app.route('/variables')
+    def get_variables():
+        return jsonify(data['variables_description'])
+
+    @app.route('/variable/<id>')
+    def get_variable(id):
+        variable = data['variables'].get(id)
+        if variable is None:
+            raise abort(404)
+        else:
```
Comment (MattiSG) [inline openfisca_web_api/app.py]: > This `else` is not necessary since you `raise` otherwise 🙂

File: openfisca_web_api/loader/variables.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import textwrap
+
+from commons import get_next_day
+
+
+def format_value(value):
+    if not isinstance(value, datetime.date):
+        return value
+    else:
+        return value.isoformat()
```
Comment (MattiSG) [inline openfisca_web_api/loader/variables.py]: > I would recommend treating special cases first (`if isinstance return do_something`) and ending with a `return value` outside of any `if` block. This scales more.

File: openfisca_web_api/loader/variables.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+
+def build_source_url(country_package_metadata, source_file_path, start_line_number, source_code):
+    nb_lines = source_code.count('\n')
+    return '{}/blob/{}{}#L{}-L{}'.format(
+        country_package_metadata['repository_url'],
+        country_package_metadata['version'],
```
Comment (MattiSG) [inline openfisca_web_api/loader/variables.py]: 👏

File: tests/test_variables.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+def check_input_variable_value(key, expected_value):
+    assert_equal(input_variable[key], expected_value)
+
+
+def test_input_variable_value():
+    expected_values = {
+        u'description': u'Date de naissance',
+        u'valueType': u'Date',
+        u'defaultValue': u'1970-01-01',
+        u'definitionPeriod': u'eternity',
```
Comment (MattiSG) [inline tests/test_variables.py]: What about using `ETERNITY` (all caps) for similarity with the constant when writing a formula?

Comment (fpagnoux) [inline tests/test_variables.py]: Why not. Should we use `MONTH` and `YEAR` ?

**Pull request merged**